examples/data_preprocessing/[WIP]_chexpert_dataset/prepare_CheXpert_data.py

```diff
@@ -0,0 +1,243 @@
+The CheXpert test set is not publicly available, as it is used for the CheXpert competition (see link above). 
+The reports that were used to label the images are also unavailable.
+"""
+
+## Imports
+
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+import torch
+import os
+import torchvision.transforms as transforms
+import joblib
+from PIL import Image
+from torch.utils.data import Dataset, DataLoader
+from tqdm import tqdm
+from tabulate import tabulate
+
+## Loading the dataset
+
+"""
+read the dataset; replace PATH with the path to the folder in which you stored train.csv and valid.csv
+if you did not change the folder structure, this path should end with "\CheXpert-v1.0-small\CheXpert-v1.0-small"
+"""
+
+path = PATH
+os.chdir(path) #change working directory to appropriate location
+print(path)
+training_set = pd.read_csv("train.csv")
+
+validation_set = pd.read_csv("valid.csv")
+
+"""
+first 5 entries of training set
+note that this dataset has 4 possible labels:
+    positive (1.0)
+    negative (0.0)
+    uncertain (-1.0)
+    not-mentioned (blank), which is read in as NaN
+223414 observations in entire training set
+"""
+training_set.head(5)
+
+"""
+first 5 entries of validation set
+note that this dataset has 2 possible labels:
+    positive (1.0)
+    negative (0.0)
+234 observations in entire validation set
+"""
+validation_set.head(5)
+
+"""
+in the following we look at some dataset statistics
+"""
+
+training_labels = training_set.iloc[:, -13:-1]
+labels_per_row = training_labels.count(axis = 1) #number of non-NaN labels per row in the training set
+
+vals = pd.DataFrame(labels_per_row.value_counts())
+
+#make a table
+val_list = []
+for i in vals.index:
+    val_list.append([i, vals[0][i]])
+    
+print(tabulate(val_list, headers = ["Number of non-NaN labels", "Number of datapoints"]))
+
+#label distribution in training set
+val_list = []
+for cond in training_labels.columns:
+    vals = np.array(training_labels.value_counts(subset=[cond]).sort_index(ascending = True))
+    val_list.append([cond, vals[0], vals[1], vals[2]])
+
+print("Label distribution in the training set:", "\n")
+print(tabulate(val_list, headers = ["Pathology", "-1.0", "0.0", "1.0"]))
+
+#label distribution in validation set
+validation_labels = validation_set.iloc[:, -13:-1]
+
+val_list = []
+for cond in validation_labels.columns:
+    vals = np.array(validation_labels.value_counts(subset=[cond]).sort_index(ascending=True))
+    if cond != "Fracture":
+        val_list.append([cond, vals[0], vals[1]])
+    else:
+        val_list.append([cond, vals[0], 0]) #fracture never positively appears in validation set
+        
+print("Label distribution in the validation set:", "\n")
+print(tabulate(val_list, headers = ["Pathology", "0.0", "1.0"]))
+
+#sample image from training set
+image_paths_sample = path[: path.find("CheXpert-v1.0-small")] + "CheXpert-v1.0-small/" + training_set["Path"]

```

Comment:
when you create a new path, it is always better to work with `os.path.join` (see https://docs.python.org/3/library/os.path.html#os.path.join), since e.g. "/" will not work for OSX/Linux but won't work for Windwos and vice versa